components/prism-cpp.js

```diff
@@ -1,6 +1,6 @@
 Prism.languages.cpp = Prism.languages.extend('c', {
 	'class-name': {
-		pattern: /(\b(?:class|enum|struct)\s+)\w+/,
+		pattern: /(\b(?:class|enum class|enum|struct)\s+)\w+/,
```

Comment:
Rather than this solution, I think it's better to disallow keyword class names.
```suggestion
		pattern: /(\b(?:class|enum|struct)\s+)(?!class|enum|struct)\w+/,
```